src/module.c

```diff
@@ -5905,6 +5962,16 @@ robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int
  *              If everything succeeded, it will return with a NULL, otherwise it will
  *              return with a CallReply object denoting the error, as if it was called with
  *              the 'E' code.
+ *     * 'K' -- Allow running blocking commands. If enabled and the command gets blocked, a
+ *              special REDISMODULE_REPLY_PROMISE will be returned. This reply type
+ *              indicates that the command was blocked and the reply will be given asynchronously.
+ *              The module can use this reply object to set a handler which will be called when
+ *              the command gets unblocked using RedisModule_CallReplyPromiseSetUnblockHandler.
+ *              If the handler is not set, the blocking command will still continue its execution
+ *              but the reply will be ignored (fire and forget). The handler must be set immediately
+ *              after the command invocation (without releasing the Redis lock in between).
+ *              The module should not keep the promise call reply after the Redis lock has been released.
+ *              The module should not free the promise call reply.
```

Comment:
@MeirShpilraien We need to document here what is allowed and not allowed in the context of the unblock handler. Because we took the permissive approach to the API and provided a RedisModuleCtx, we need to provide this guidance at the documentation level.